Tests/Tests/STPFormEncoderTest.m

```diff
@@ -98,6 +98,21 @@ - (void)testFormEncoding_arrayValue {
     XCTAssertEqualObjects([self encodeObject:testObject], @"test_object[test_array_property][]=1&test_object[test_array_property][]=2&test_object[test_array_property][]=3&test_object[test_property]=success");
 }
 
+- (void)testFormEncoding_arrayOfEncodable {
+    STPTestFormEncodableObject *testObject = [STPTestFormEncodableObject new];
+
+    STPTestFormEncodableObject *inner1 = [STPTestFormEncodableObject new];
+    inner1.testProperty = @"inner1";
+    STPTestFormEncodableObject *inner2 = [STPTestFormEncodableObject new];
+    inner2.testArrayProperty = @[@"inner2"];
+
+    testObject.testArrayProperty = @[inner1, inner2];
+
+    XCTAssertEqualObjects([self encodeObject:testObject],
+                          @"test_object[test_array_property][0][test_property]=inner1"
+                          "&test_object[test_array_property][1][test_array_property][0]=inner2");
```

Comment:
I'm not sure if `[0]` and `[1]` are the right solution here. Still need to test against the Stripe API.

Reply:
Talked to basta & tested against the API, this *is* correct.